test/tracer.js

```diff
@@ -0,0 +1,230 @@
+        tags[`${opentracing_tags.SPAN_KIND}`] = opentracing_tags.SPAN_KIND_RPC_SERVER;
+
+        let childOfParams = {
+            operationName: 'test-name',
+            childOf: context,
+            startTime: startTime,
+            tags
+        }
+
+        let referenceParams = {
+            operationName: 'test-name',
+            references: [new opentracing.Reference(opentracing.REFERENCE_CHILD_OF, context)],
+            startTime: startTime,
+            tags
+        }
+
+        let assertByStartSpanParameters = (params) => {
+            let span = tracer.startSpan(params);
+
+            assert.isOk(bufferEqual(span.context().traceId, traceId));
+            assert.isOk(bufferEqual(span.context().parentId, parentId));
+            assert.equal(span.context().flags, constants.SAMPLED_MASK);
+            assert.equal(span._start, startTime);
+        }
+
+        assertByStartSpanParameters(childOfParams);
+        assertByStartSpanParameters(referenceParams);
+    });
+
+    it ('inject plain text headers into carrier, and extract span context with the same value', () => {
+        let keyOne = 'keyOne';
+        let keyTwo = 'keyTwo';
+        let baggage = {
+            keyOne: 'leela',
+            keyTwo: 'bender'
+        };
+        let savedContext = new SpanContext(
+            Utils.encodeInt64(1),
+            Utils.encodeInt64(2),
+            Utils.encodeInt64(3),
+            constants.SAMPLED_MASK,
+            baggage
+        );
+
+        let assertByFormat = (format) => {
+            let carrier = {};
+            tracer.inject(savedContext, format, carrier);
+            let extractedContext = tracer.extract(format, carrier);
+
+            assert.isOk(bufferEqual(savedContext.traceId, extractedContext.traceId));
+            assert.isOk(bufferEqual(savedContext.spanId, extractedContext.spanId));
+            assert.isOk(bufferEqual(savedContext.parentId, extractedContext.parentId));
+            assert.equal(savedContext.flags, extractedContext.flags);
+            assert.equal(savedContext.baggage[keyOne], extractedContext.baggage[keyOne]);
+            assert.equal(savedContext.baggage[keyTwo], extractedContext.baggage[keyTwo]);
+        }
+
+        assertByFormat(opentracing.FORMAT_TEXT_MAP);
+        assertByFormat(opentracing.FORMAT_HTTP_HEADERS);
+    });
+
+    it ('inject url encoded values into headers', () => {
+        let baggage = {
+            keyOne: 'leela',
+            keyTwo: 'bender'
+        };
+        let savedContext = new SpanContext(
+            Utils.encodeInt64(1),
+            Utils.encodeInt64(2),
+            Utils.encodeInt64(3),
+            constants.SAMPLED_MASK,
+            baggage
+        );
+        let carrier = {};
+
+        tracer.inject(savedContext, opentracing.FORMAT_HTTP_HEADERS, carrier);
+        assert.equal(carrier['UberCtx-keyOne'], encodeURIComponent('leela'));
+        assert.equal(carrier['UberCtx-keyTwo'], encodeURIComponent('bender'));
```

Comment:
this validation is moot, because you picked the values that do not change when passed through url encoding.

Try values with spaces like `"Leela vs. Bender"` and verify that spaces will be represented as `%20`. Compare with literal expected string, not by delegating to encodeURI method again.